cpp/src/jni/exec_backend.h

```diff
@@ -46,8 +47,10 @@ class ResultIteratorBase {
 class ExecBackendBase : public std::enable_shared_from_this<ExecBackendBase> {
  public:
   virtual ~ExecBackendBase() = default;
-  virtual std::shared_ptr<ArrowArrayResultIterator> GetResultIterator() = 0;
   virtual std::shared_ptr<ArrowArrayResultIterator> GetResultIterator(
+      gluten::memory::MemoryAllocator* allocator) = 0;
```

Comment:
The backend interface has changed. Could you also make the changes in gazelle_cpp?

Reply:
done.